applications/crossbar/src/cb_context.erl

```diff
@@ -1233,3 +1236,30 @@ build_error_message(_Version, Message) when is_binary(Message) ->
     wh_json:from_list([{<<"message">>, Message}]);
 build_error_message(_Version, JObj) ->
     JObj.
+
+-spec maybe_convert_numbers(cb_context:context(), wh_json:object(), jesse_error:error_reasons()) -> cb_context:context().
+maybe_convert_numbers(Context, SchemaJObj, Errors) ->
+    JObj = req_data(Context),
+    case lists:foldl(fun maybe_convert_number/2, JObj, Errors) of
+        JObj ->
+            lager:debug("request data did not validate against ~s: ~p", [wh_doc:id(SchemaJObj)
+                                                                         ,Errors
+                                                                        ]),
+            failed(Context, Errors);
+        NewJObj ->
+            validate_request_data(SchemaJObj, set_req_data(Context, NewJObj))
+    end.
+
+-spec maybe_convert_number(jesse_error:error_reason(), wh_json:object()) -> wh_json:object().
+maybe_convert_number({data_invalid, {Props}, 'wrong_type', Value, Key}, JObj) ->
```

Comment:
Why pattern match the tuple? This is a JObj, no? You can use `wh_json` below to get the type.